test/sqllogictest/list.slt

```diff
@@ -2575,3 +2575,244 @@ SELECT LIST[LIST[customer.first_name, customer.last_name], LIST[customer.zip], L
 ----
 {{alice,lasta},{10003},{1}}
 {{charlie,lastc},{11217},{3}}
+
+## Optimize list_index(list_create, literal) and multidimensional variations
+# Most of these tests could also be in src/expr/tests/testdata/reduce, but writing these big ListCreate expressions is a
+# bit cumbersome in json.
+
+statement ok
+CREATE TABLE t3(f1 int, f2 int, f3 int, f4 int, f5 int, f6 int, f7 int, f8 int, n int, m int, l int list)
+
+statement ok
+INSERT INTO t3 VALUES (1, 2, 3, 4, 5, 6, 7, 8, 1, 2, list[42, 43])
+
```

Comment:
Could you please add more comments to this file describing the different scenarios being tested?

Reply:
Done